code/modules/mob/living/carbon/human/species_types.dm

```diff
@@ -622,6 +740,229 @@
 			active = null
 	..()
 
+//Immune to physical bullets and resistant to brute, but very vulnerable to burn damage. Dusts on death.
+/datum/species/golem/sand
+	name = "Sand Golem"
+	id = "sand"
+	fixed_mut_color = "ffdc8f"
+	meat = /obj/item/weapon/ore/glass //this is sand
+	armor = 0
+	burnmod = 3 //melts easily
+	brutemod = 0.25
+	info_text = "As a <span class='danger'>Sand Golem</span>, you are immune to physical bullets and take very little brute damage, but are extremely vulnerable to burn damage. You will also turn to sand when dying, preventing any form of recovery."
+
+/datum/species/golem/sand/spec_death(gibbed, mob/living/carbon/human/H)
+	H.visible_message("<span class='danger'>[H] turns into a pile of sand!</span>")
+	for(var/obj/item/W in H)
+		H.unEquip(W)
+	for(var/i=1, i <= rand(3,5), i++)
+		new /obj/item/weapon/ore/glass(get_turf(H))
+	qdel(H)
+
+/datum/species/golem/sand/bullet_act(obj/item/projectile/P, mob/living/carbon/human/H)
+	if(!(P.original == H && P.firer == H))
+		if(istype(P, /obj/item/projectile/bullet) || istype(P, /obj/item/projectile/kinetic))
+			playsound(H, "sound/effects/shovel_dig.ogg", 70, 1)
+			H.visible_message("<span class='danger'>The [P.name] sinks harmlessly in [H]'s sandy body!</span>", \
+			"<span class='userdanger'>The [P.name] sinks harmlessly in [H]'s sandy body!</span>")
+			return 2
+	return 0
+
+//Reflects lasers and resistant to burn damage, but very vulnerable to brute damage. Shatters on death.
+/datum/species/golem/glass
+	name = "Glass Golem"
+	id = "glass"
+	fixed_mut_color = "5a96b4aa" //transparent body
+	meat = /obj/item/weapon/shard
+	armor = 0
+	brutemod = 3 //very fragile
+	burnmod = 0.25
+	info_text = "As a <span class='danger'>Glass Golem</span>, you reflect lasers and energy weapons, and are very resistant to burn damage, but you are extremely vulnerable to brute damage. On death, you'll shatter beyond any hope of recovery."
+
+/datum/species/golem/glass/spec_death(gibbed, mob/living/carbon/human/H)
+	playsound(H, "shatter", 70, 1)
+	H.visible_message("<span class='danger'>[H] shatters!</span>")
+	for(var/obj/item/W in H)
+		H.unEquip(W)
+	for(var/i=1, i <= rand(3,5), i++)
+		new /obj/item/weapon/shard(get_turf(H))
+	qdel(H)
+
+/datum/species/golem/glass/bullet_act(obj/item/projectile/P, mob/living/carbon/human/H)
+	if(!(P.original == H && P.firer == H)) //self-shots don't reflect
+		if(istype(P, /obj/item/projectile/beam) || istype(P, /obj/item/projectile/energy))
```

Comment:
This should check the projectile's damage flag(the `flag` var) instead of types, ie; `if(P.flag == "laser" || P.flag == "energy")`